### DISCUSSION TITLE
Change serving size when adding meals to the shopping list or meal plan

### DISCUSSION THREAD

**Discussion by Belgadon** [mealie-recipes/mealie#2303] [Feature Request]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [X] Reviewed the documentation and verified that my feature request isn't already a feature.
- [X] Reviewed the existing feature requests and my feature has not been requested.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
From recipe overview page you can add recipes to shopping list only with the default serving size (e.g. 1). To do so, go to the recipe overview/search page, click on the three dots on a recipe tile, choose "add to shopping list" and a shopping list.
A modal/popup appears where one can select/deselect certain ingredients. If one deselects one ingredient (e.g. water or salt) it is not added to the shopping list/meal plan.
When one later changes the serving size, the deselected ingredient gets add to the list for the newly added serving.

**Suggest A Solution**
When adding a recipe to the shopping list or meal plan one should be able to change the serving size in the modal/popup.
![1](https://user-images.githubusercontent.com/36801399/228481197-f3ea4ea1-c4f7-49ef-8a9a-e51b60869658.PNG)
![2](https://user-images.githubusercontent.com/36801399/228481201-393ce73a-ab53-4dd5-b990-fd3f9f461498.PNG)

**Additional Information**
- [X] If the feature is accepted I'm willing to TRY to submit a PR to provide this feature ;-)

**Comment by Belgadon** [comment]
The more I think about it and the more I play around in the development environment, the clearer the picture of what could be really helpful gets.

I think this is closely related to #1899 .
Perhaps the serving size should be selected from the menu planner view when adding to the shopping list. And not when it's added to the meal planner.

That said, if one adds a recipe to a shopping list directly from the contex menu, one should still be able to choose the serving size.

**Comment by michael-genson** [comment]
This is partially available already; if you change the scale when viewing the recipe, that scale is added to your shopping list.

I agree it should be customizable in the dialog itself, too. It's also still not possible in the meal plan

**Reply by Yowaishi** [reply]
Just want to add my 2 cents - it would be nice if the linked recipes on the shopping list allowed you adjust how many servings rather than just doubling/tripling the recipe.

 I have recipes that default to 2 servings, or 4 servings, others to 12, but I have no idea which is which when just looking at the shopping list. Currently I just go to my meal plan and click into each recipe individually, adjust number of servings, add to shopping list, then move onto the next recipe. this is even with the just released feature of 'all all to list' on the meal plan page.

**Comment by sparkison** [comment]
I've created a PR to allow adjusting the serving size within the meal planner (works the same as it does on the recipe detail page), which should solve this: https://github.com/mealie-recipes/mealie/pull/7401

The serving size is respected when clicking into the recipe as well, so the recipe detail page will respect the serving size set in the meal plan. Same is true when adding to a shopping list.

This has been tested on my own server, as well as my local test instance. It's working great!

<img width="1020" height="461" alt="scale-in-menu-planner" src="https://github.com/user-attachments/assets/f0057f43-3e7e-47a6-a08f-2b8b205f4cd6" />
